syntax = "proto3";

package nighthawk.adaptive_load;

import "envoy/config/core/v3/extension.proto";
import "google/rpc/status.proto";

// Configuration for FakeStepController (plugin name: "nighthawk.fake_step_controller") that always
// returns a fixed RPS value and changes converged and doomed states based on the latest reported
// BenchmarkResult.
message FakeStepControllerConfig {
  // RPS that should always be returned, except when artificial errors are configured. Optional,
  // default 0.
  int32 fixed_rps_value = 1;
  // Artificial error that the plugin factory should return during validation. Optional.
  google.rpc.Status artificial_validation_failure = 2;
  // Artificial error that should be returned from GetCurrentCommandLineOptions(). Optional.
  google.rpc.Status artificial_input_setting_failure = 3;
}
